src/execution/riscv64/simulator-riscv64.cc

```diff
@@ -1268,7 +1268,7 @@ T Simulator::ReadMem(int64_t addr, Instruction* instr) {
   }
 
   // check for natural alignment
-  if ((addr & (sizeof(T) - 1)) != 0) {
+  if (((addr & 1) != 0) && ((addr & (sizeof(T) - 1)) != 0)) {
```

Comment:
Why add 'addr & 1' ? If add it '((addr & (sizeof(T) - 1)) != 0)' will make no sense.

Reply:
The first is the case were sizeof(T) > 1 ie 4 byte loads. The second is the original code and is there for 1 byte loads. (addr & 1) does not work for the latter because (addr & (sizeof(T)-1)) evaluates to (addr & 0) when sizeof(T) is one. Please let me know if you would like the logic written another way.

Reply:
Why the 0 bit must zero when load a byte ? According to my understanding, load a byte don't need align?

Reply:
Yes that is correct. With C-Extension I think that none of the loading needs alignment. So when you put it that way I guess I can remove the whole if statement there.